spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDParser.java

```diff
@@ -43,4 +45,29 @@
 	 *         described by the DSL.
 	 */
 	List<ModuleDescriptor> parse(String name, String config, ParsingContext type);
+
+	public static class DefinitionOrException {
```

Comment:
I would disagree, as the role of the controller should be merely to interface between the HTTP world and business layer. If we were to switch transports, nothing should have to be rewritten.

Moving the loop upwards would require to move the transient repository thing there as well, clearly leaking business logic in the controller IMO.

I only see two ways to convey a mixed result: either via the `return` keyword, as the current state of this PR does, or via the `throw` keyword (we could have e.g. a PartialParsingFailureException, but then there would be code duplication with the 100% happy case)